packages/crowdstrike/data_stream/fdr/elasticsearch/ingest_pipeline/default.yml

```
@@ -2227,17 +2227,22 @@ processors:
   - set:
       field: dns.type
       value: query
-      if: ctx.event?.action == "DnsRequest"
+      if: ctx.event?.action != null && ctx.event?.action.contains("DnsRequest")
   - registered_domain:
       field: crowdstrike.DomainName
       target_field: dns.question
       ignore_missing: true
-      if: ctx.event?.action == "DnsRequest"
+      if: ctx.event?.action != null && ctx.event?.action.contains("DnsRequest")
```

```suggestion
      if: ctx.event?.action != null && ctx.event.action.contains("DnsRequest")
```
Though could we use `ctx.dns?.type == "query"` as the condition?